nalgebra-glm/src/ext/matrix_clip_space.rs

```diff
@@ -53,119 +53,1200 @@ use na::{Orthographic3, Perspective3, Real};
-//    unimplemented!()
-//}
-//
-//pub fn ortho_rh_no<N: Real>(left: N, right: N, bottom: N, top: N, znear: N, zfar: N) -> TMat4<N> {
-//    unimplemented!()
-//}
-//
-//pub fn ortho_rh_zo<N: Real>(left: N, right: N, bottom: N, top: N, znear: N, zfar: N) -> TMat4<N> {
-//    unimplemented!()
-//}
-//
-//pub fn ortho_zo<N: Real>(left: N, right: N, bottom: N, top: N, znear: N, zfar: N) -> TMat4<N> {
-//    unimplemented!()
-//}
+/// Creates a left hand matrix for a orthographic-view frustum with a depth range defined by the
+/// default configured for the library at build time
+///
+/// # Parameters
+///
+/// * `left` - Coordinate for left bound of matrix
+/// * `right` - Coordinate for right bound of matrix
+/// * `bottom` - Coordinate for bottom bound of matrix
+/// * `top` - Coordinate for top bound of matrix
+/// * `znear` - Distance from the viewer to the near clipping plane
+/// * `zfar` - Distance from the viewer to the far clipping plane
+///
+/// # Compile Options
+///
+/// There are 2 compile options that change the behaviour of the function:
+/// 1. projection_y_flip
+/// 2. zero_to_one_clip_default/negone_to_one_clip_default
+///
+/// ##### projection_y_flip
+/// When enabled will perform a `matrix[(1,1)] *= 1` implicitly. Primary use case is for Vulkan
+/// where the viewport coordinate origin is the top left, unlike OpenGL which is the bottom left.
+///
+/// ##### zero_to_one_clip_default/negone_to_one_clip_default
+/// Depending on which option is set the function will return a orthographic matrix meant for a
+/// 0 to 1 depth clip space or a -1 to 1 depth clip space.
+///
+pub fn ortho_lh<N: Real>(left: N, right: N, bottom: N, top: N, znear: N, zfar: N) -> TMat4<N> {
+    if cfg!(feature="zero_to_one_clip_default") {
+        ortho_zo(left, right, bottom, top, znear, zfar)
+    } else if cfg!(feature="negone_to_one_clip_default") {
+        ortho_no(left, right, bottom, top, znear, zfar)
+    } else {
+        unimplemented!()
+    }
+}
+
+/// Creates a left hand matrix for a orthographic-view frustum with a depth range of -1 to 1
+///
+/// # Parameters
+///
+/// * `left` - Coordinate for left bound of matrix
+/// * `right` - Coordinate for right bound of matrix
+/// * `bottom` - Coordinate for bottom bound of matrix
+/// * `top` - Coordinate for top bound of matrix
+/// * `znear` - Distance from the viewer to the near clipping plane
+/// * `zfar` - Distance from the viewer to the far clipping plane
+///
+/// # Compile Options
+///
+/// There are 1 compile option that changes the behaviour of the function:
+/// 1. projection_y_flip
+///
+/// ##### projection_y_flip
+/// When enabled will perform a `matrix[(1,1)] *= 1` implicitly. Primary use case is for Vulkan
+/// where the viewport coordinate origin is the top left, unlike OpenGL which is the bottom left.
+///
+pub fn ortho_lh_no<N: Real>(left: N, right: N, bottom: N, top: N, znear: N, zfar: N) -> TMat4<N> {
+    let zero   : N = ::convert(0.0);
```

Comment:
This should be `let zero = N::zero();`. The same remark applies to other occurrences bellow.